### ISSUE TITLE
Activated FRN

### ISSUE THREAD

**Issue by qubvel** [tattaka/Filter-Response-Normalization-PyTorch#2]
I have modified code, what do you think about it?
Activated version of FRN simplify model conversion

```python
class _FilterResponseNorm(Module):
    __constants__ = ["num_features", "eps", "eps_trainable", "tau", "beta", "gamma"]

    def __init__(self, shape, activated=True, eps=1e-6, eps_trainable=True):
        super(_FilterResponseNorm, self).__init__()
        self._eps = eps
        self.activated = activated
        self.num_features = shape[1]
        self.eps_trainable = eps_trainable

        self.beta = Parameter(torch.zeros(shape))
        self.gamma = Parameter(torch.ones(shape))

        if self.eps_trainable:
            self.eps = Parameter(torch.full(shape, eps))
        else:
            self.eps = eps

        if self.activated:
            self.tau = Parameter(torch.zeros(shape))
        else:
            self.tau = None

        self.reset_parameters()

    def reset_parameters(self):
        init.zeros_(self.beta)
        init.ones_(self.gamma)
        if isinstance(self.eps, Parameter):
            init.constant_(self.eps, self._eps)
        if self.tau is not None:
            init.zeros_(self.tau)

    def _check_input_dim(self, input):
        raise NotImplementedError

    def forward(self, input):
        self._check_input_dim(input)
        nu2 = torch.mean(input ** 2, axis=2, keepdims=True)
        input = input * torch.rsqrt(nu2 + torch.abs(self.eps) + self._eps)
        output = self.gamma * input + self.beta
        if self.activated:
            output = threshold_linear_unit(output, self.tau)
        return output


class FilterResponseNorm1d(_FilterResponseNorm):

    def __init__(self, num_features, activated=True, eps=1e-6, eps_trainable=True):
        super(FilterResponseNorm1d, self).__init__(
            shape=(1, num_features, 1),
            activated=activated,
            eps=eps,
            eps_trainable=eps_trainable,
        )

    def _check_input_dim(self, input):
        if input.dim() != 3:
            raise ValueError('expected 4D input (got {}D input)'
                             .format(input.dim()))


class FilterResponseNorm2d(_FilterResponseNorm):

    def __init__(self, num_features, activated=True, eps=1e-6, eps_trainable=True):
        super(FilterResponseNorm2d, self).__init__(
            shape=(1, num_features, 1, 1),
            activated=activated,
            eps=eps,
            eps_trainable=eps_trainable,
        )

    def _check_input_dim(self, input):
        if input.dim() != 4:
            raise ValueError('expected 4D input (got {}D input)'
                             .format(input.dim()))

def convert_model(module):
    """Traverse the input module and its child recursively
       and replace all instance of torch.nn.modules.batchnorm.BatchNorm*N*d
       to SynchronizedBatchNorm*N*d
    Args:
        module: the input module needs to be convert to FRN model
    Examples:
        >>> import torch.nn as nn
        >>> import torchvision
        >>> # m is a standard pytorch model
        >>> m = torchvision.models.resnet18(True)
        >>> m = nn.DataParallel(m)
        >>> # after convert, m is using FRN
        >>> m = convert_model(m)
    """

    mod = module
    if isinstance(module, torch.nn.modules.batchnorm.BatchNorm1d):
        mod = FilterResponseNorm1d(module.num_features, activated=True, eps=module.eps)
    if isinstance(module, torch.nn.modules.batchnorm.BatchNorm2d):
        mod = FilterResponseNorm2d(module.num_features, activated=True, eps=module.eps)
    elif isinstance(module, torch.nn.ReLU):
        mod = torch.nn.Identity()
    for name, child in module.named_children():
        mod.add_module(name, convert_model(child))
    return mod
```

**Comment by tattaka** [comment]
This is a very cool proposal. Thank you, I will implement it.

**Comment by tattaka** [comment]
I pushed because I passed the test. Thank you very much !!
